Formula/ugit.rb

```diff
@@ -0,0 +1,20 @@
+class Ugit < Formula
+  desc "🚨️ ugit helps you undo git commands. Your damage control git buddy"
+  homepage "https://github.com/Bhupesh-V/ugit"
+  url "https://github.com/Bhupesh-V/ugit/archive/refs/tags/v5.0.tar.gz"
+  sha256 "f1d603756635675b557cb43eda01443b87f5024aab455283f06a3fd9cfbf2d06"
+  license "MIT"
+
+  depends_on "bash"
+  depends_on "fzf"
+
+  def install
+    bin.install "ugit"
+    bin.install "git-undo"
+  end
+
+  test do
+    assert_match "ugit version", shell_output("#{bin}/ugit --version")
+    assert_match "ugit help", shell_output("#{bin}/ugit --help")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
It's hard to write tests for ugit now since it doesn't really manual input from a user
Writing a test for creating and reading its output doesn't feel useful either
Could you direct me to a different approach that could help?

Reply:
>  Writing a test for creating and reading its output doesn't feel useful either

Since that checks if it can actually do what it should, that sounds pretty good to me.